tests/queries/test_returning.py

```diff
@@ -0,0 +1,29 @@
+import unittest
+
+from django.db import connection, router
+from django.db.models.sql import InsertQuery
+from django.test import TestCase
+
+from .models import Author, ReturningModel
+
+
+@unittest.skipUnless(connection.vendor == 'postgresql', 'PostgreSQL tests')
```

Comment:
Shouldn't this be gated by a feature flag instead since it's supposed to work on Oracle as well?

Isn't this missing actual tests for `ReturningModel` inserts?

Reply:
HI @charettes no, it does not work on oracle, even tho the code suggest it. It is actually not supported by the db client as far as I understood it :/
I'd love to see this work in the future, for now this remains PG only.

Reply:
So all the other code is just not NOT break Oracle and I added support for non-int primary keys, as the docs suggest.

Reply:
I still think that this test should execute the SQL it generates to assert it's working. Right now it's only introspecting it which doesn't add much value.

To be clear I think that the only thing that should be tested here is that `ReturningModel.objects.create()` has both an `id` and `isinstance(r.created, datetime)` assigned.